### PR TITLE
Always define 'format' target

### PR DESCRIPTION
**Pull request by mkindahl**
If `clang-format` is not found `CLANG_FORMAT` will be set to a false
value, which will prevent the `format` target from being defined if
`cmake-format` is not installed.

This commit fixes that by always creating a `format` target since
`clang-format` is always created even if `CLANG_FORMAT` is false.

We do not redefine `CLANG_FORMAT` to a non-false value since it is
expected to contain a path to an executable and if set to a true value
that is not a path it could be used in a way that leads to strange
errors.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -440,12 +440,12 @@ if(CMAKE_FORMAT)
                          ${PROJECT_SOURCE_DIR}/scripts/cmake_format_all.sh)
 endif()
 
-if(CMAKE_FORMAT OR CLANG_FORMAT)
-  add_custom_target(format)
-  add_dependencies(format clang-format)
-  if(CMAKE_FORMAT)
-    add_dependencies(format cmake-format)
-  endif()
+# We define the target 'format' all the time since 'clang-format' is guaranteed
```
Comment (mfundul) [inline CMakeLists.txt:443]:
```suggestion
# We always define the target 'format' since 'clang-format' is guaranteed
```

**Review by mfundul — APPROVED** [review]

**Review by nikkhils — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 2]
```diff
@@ -440,12 +440,12 @@ if(CMAKE_FORMAT)
                          ${PROJECT_SOURCE_DIR}/scripts/cmake_format_all.sh)
 endif()
 
-if(CMAKE_FORMAT OR CLANG_FORMAT)
-  add_custom_target(format)
-  add_dependencies(format clang-format)
-  if(CMAKE_FORMAT)
-    add_dependencies(format cmake-format)
-  endif()
+# We always define the target 'format' since 'clang-format' is guaranteed to
```
Comment (afiskon) [inline CMakeLists.txt:443]: What do you mean by 'clang-format is guaranteed to exist'? I'm afraid this is not true. This comment needs a little more context.

Comment (afiskon) [reply CMakeLists.txt:443]: Oh.... after reading it like 5 times I realized you probably meant the target, not the executable. This is not obvious.

**Review by afiskon — APPROVED** [review]

**Pull request merged**